core/src/actors/chain_manager/handlers.rs

```diff
@@ -85,8 +85,11 @@ impl Handler<EpochNotification<EveryEpochPayload>> for ChainManager {
                         self.current_epoch.unwrap(),
                     );
 
-                    for _reveal in reveals {
-                        // FIXME(#337): broadcast transaction
+                    for reveal in reveals {
+                        // Send AddTransaction message to self
+                        // This will run all the validations again
```

Comment:
Sounds sub-optimal to me, unless we make sure we don't do any work twice.

May we put a TODO here in order to leave some trace for us to revisit this algorithm later?